flink-end-to-end-tests/flink-avro-classloading-test/src/main/avro/user.avsc

```diff
@@ -0,0 +1,9 @@
+{
+     "type": "record",
+     "namespace": "org.apache.flink.tests.streaming",
+     "name": "User",
+     "fields": [
+       { "name": "first", "type": "string" },
+       { "name": "last", "type": "string" }
```

Comment:
Could you add a more complex Avro type? I think we should test all available Avro types. Esp. maps and arrays and maybe different `SpecificRecords` and `GenericRecords`.

Reply:
I can do that. The initial idea was to add this e2e test for a specific failure case that got fixed in 1.4 something, but there shouldn't be a downside to extending it with more complex fields, should it?

Reply:
I added some more complex examples and verified that the initial test case still fails on 1.3.X and works on 1.4.X as expected